webrtc-stats.html

```diff
@@ -1294,10 +1297,106 @@ <h2>
                   CNAME item.
                 </p>
               </dd>
+              <dt>
+                <dfn><code>qualityLimitationReason</code></dfn> of type <span class=
+                "idlMemberType"><a>RTCQualityLimitationReason</a></span>
+              </dt>
+              <dd>
+                <p>
+                  Only valid for video. The current reason for limiting the resolution and/or
+                  framerate, or "none" if not limited.
+                </p>
+              </dd>
+              <dt>
+                <dfn><code>qualityLimitationDurations</code></dfn> of type <span class=
+                "idlMemberType"><a>record&lt;RTCQualityLimitationReason, double&gt;</a></span>
+              </dt>
+              <dd>
+                <p>
+                  A record of the total time, in seconds, that this stream has spent being
+                  resolution and/or framerate limited for the reason specified by the record's
+                  key.
+                </p>
+              </dd>
+              <dt>
+                <dfn><code>qualityTotalLimitationDuration</code></dfn> of type <span class=
+                "idlMemberType"><a>double</a></span>
+              </dt>
+              <dd>
+                <p>
+                  The total time, in seconds, that this stream has spent being resolution and/or
+                  framerate limited (that <code>qualityLimitationReason</code> has had a value
+                  other than "none").
+                </p>
+              </dd>
             </dl>
           </section>
         </div>
       </section>
+      <section>
+        <h3>
+          <dfn>RTCQualityLimitationReason</dfn> enum
+        </h3>
+        <div>
+          <pre class="idl">enum RTCQualityLimitationReason {
+            "none",
+            "cpu",
+            "bandwidth",
+            "other",
+          };</pre>
+          <table data-link-for="RTCQualityLimitationReason" data-dfn-for="RTCQualityLimitationReason" class="simple">
+            <tbody>
+              <tr>
+                <th colspan="2">
+                  Enumeration description
+                </th>
+              </tr>
+              <tr>
+                <td>
+                  <dfn><code>none</code></dfn>
+                </td>
+                <td>
+                  <p>
+                    The resolution and/or framerate is not limited.
```

Comment:
I think it's clearer if we make it such that none must be filled in if implemented or have a statement here that none shouldn't be filled in and the base for the time to use for this is the track time (or what the appropriate time to use as the difference).

I agree that having nearly any sane standard is better then having it up to each implementation.

Reply:
For "RTCQualityLimitationReason qualityLimitationReason" I explicitly state "or "none" if not limited". Only if not implemented or applicable would it be undefined (as all stats in this document can be in that case).